tests/unit/test-util/test.file-filter.js

```diff
@@ -131,4 +131,31 @@ describe('util/file-filter', () => {
     });
   });
 
+  describe('negation', () => {
+    const filter = newFileFilter({
+      sourceDir: '/src',
+      ignoreFiles: [
+        '!node_modules/libdweb/src/**',
+      ],
+    });
+
+    it('ignore paths not captured by negation', () => {
+      assert.equal(filter.wantFile('/src/node_modules/lib/foo.js'), false);
+      assert.equal(filter.wantFile('/src/node_modules/lib'), false);
+      assert.equal(filter.wantFile('/src/node_modules/what.js'), false);
+      assert.equal(filter.wantFile('/src/node_modules/libdweb/what.js'), false);
+      assert.equal(filter.wantFile('/src/node_modules/libdweb/src.js'), false);
+      assert.equal(filter.wantFile('/src/node_modules/libdweb/src'), false);
+    });
+
+    it('includes paths captured by negation', () => {
+      assert.equal(
+        filter.wantFile('/src/node_modules/libdweb/src/lib.js'),
+        true);
+      assert.equal(
+        filter.wantFile('/src/node_modules/libdweb/src/sub/lib.js'),
```

Comment:
Please add a test for `/src/node_modules/some/other/node_modules/index.js` (I guess that the path is included because of the negation, but I would not be surprised if it got ignored anyway because of the deeper `node_modules/`.